src/Shared/ExceptionHandling.cs

```diff
@@ -167,7 +167,9 @@ internal static bool IsIoRelatedException(Exception e)
         internal static bool IsXmlException(Exception e)
         {
             return e is XmlException
-                || e is XmlSyntaxException
+#if NETFRAMEWORK
```

Comment:
nit: when I looked at this I chose to use `FEATURE_SECURITY_PERMISSIONS` which is defined on .NETFramework since that seems to be what the rest of the codebase is using.